### PR TITLE
Update rpc-tests (Don't merge)

### PR DESCRIPTION
**Pull request by djksa**
▶️ [NEW PULL REQUEST HERE](https://github.com/NAVCoin/navcoin-core/pull/405) ◀️ 

Fixed certain rpc-tests from Issue #358

### CODE REVIEW

File: qa/rpc-tests/wallet.py [hunk 1]
```diff
@@ -266,10 +268,12 @@ def run_test(self):
             raise AssertionError("Must not parse invalid amounts")
 
         try:
-            self.nodes[0].generate("2")
+            #self.nodes[0].generate("2")
+            slow_gen(self.nodes[0], "2")
```
Comment (matt-auckland) [inline qa/rpc-tests/wallet.py]: convert htis to generate

File: qa/rpc-tests/sendtoaddress.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        self.sync_all()
+        # Generate a block to confirm transactions
+        slow_gen(self.nodes[0], 1)
+        time.sleep(2)
+        self.sync_all()
+        # Assert transactions went through by validating balances
+        assert_equal(self.nodes[0].getbalance(), 59799900 + self.nodes[0].gettransaction(txid0)["fee"] + self.nodes[0].gettransaction(txid7)["fee"] + self.nodes[0].gettransaction(txid9)["fee"])
+        assert_equal(self.nodes[1].getbalance(), 1400 + self.nodes[1].gettransaction(txid1)["fee"])
+
+        # Make transactions with invalid addresses
```
Comment (matt-auckland) [inline qa/rpc-tests/sendtoaddress.py]: We don't check if we actually catch any errors, see the testing pattern I commented elsewhere on tis pr

File: qa/rpc-tests/sendtoaddress.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(), 59799900 + self.nodes[0].gettransaction(txid0)["fee"] + self.nodes[0].gettransaction(txid7)["fee"] + self.nodes[0].gettransaction(txid9)["fee"])
+        assert_equal(self.nodes[1].getbalance(), 1450 + self.nodes[1].gettransaction(txid1)["fee"])
+
+        # Make a transaction with immature coins included
+        balance = self.nodes[0].getbalance()
+        slow_gen(self.nodes[0], 1)
+        time.sleep(2)
+        self.sync_all()
+        try:
```
Comment (matt-auckland) [inline qa/rpc-tests/sendtoaddress.py]: same ssue as above

File: qa/rpc-tests/sendtoaddress.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        time.sleep(2)
+        self.sync_all()
+        slow_gen(self.nodes[1], 75)
+        self.sync_all()
+        
+        # Assert correct amount of NAV generated
+        assert_equal(self.nodes[0].getbalance(), 59800000)
+        assert_equal(self.nodes[1].getbalance(), 1250)
+        
+        # Make transactions to valid addresses
```
Comment (matt-auckland) [inline qa/rpc-tests/sendtoaddress.py]: Try checking what happens when we:
- send a negative number, e.g. `-1`
- send a string that isn't a valid number, e.g. `"five"`
- send something that has too many decimals, e.g. `0.000000001`
- send something that has the max number of decimals, e.g. `0.00000001`

**Comment by djksa** [comment]
Made some changes @matt-auckland

File: qa/rpc-tests/fundrawtransaction.py [hunk 5]
```diff
@@ -572,7 +575,14 @@ def run_test(self):
...
         #####################################################
 
-        rawtx   = "0100000000010000000000000000066a047465737400000000"
+        # NAVCoin: version number is 02000000 (4-bytes)
+        # NAVCoin: time field is 01000000 (4-bytes)
+        # NAVCoin: vin number is 00
+        # NAVCoin: vout number is 01
+        # NAVCoin: vout value is 0000000000000000 (8-bytes)
+        # NAVCoin: vout_length is 06 (1-byte)
+        # NAVCoin: OP_RETURN script is 6a0474657374
```
Comment (aguycalled) [inline qa/rpc-tests/fundrawtransaction.py]: transactions with version >= 2 include an additional strdzeel parameter

Comment (matt-auckland) [reply qa/rpc-tests/fundrawtransaction.py]: What does this mean for this test? we need to modify these bytes?

**Comment by craigmacgregor** [comment]
This PR needs updating. Resolve merge conflict and only include finished tests in this PR. eg. remove any which are WIP.

**Comment by matt-auckland** [comment]
Closing this as the codebase has moved on.
If we want to get this moving again I would advise making a new branch, checking out the half-finished tests from this branch, and then fixing up the tests.

**Pull request closed without merging**